Automattic/wp-calypso
# JP Connect: Handle User Already Connected Case in Redux Actions

This PR implements the following rough plan of action (that could be turned into a stretegy for other JPC auth notices):

* Start in `jetpack-connect-notices`: 
  * Locate `USER_IS_ALREADY_CONNECTED_TO_SITE` case, remove.
  * Locate what selector was used to determine that case: `getUserAlreadyConnected`. Remove from `connect`.
* Remove that selector and its unit test.
* Make note what part of the state this was using: `state.jetpackConnect.jetpackConnectAuthorize.userAlreadyConnected`. Locate corresponding [reducer](https://github.com/Automattic/wp-calypso/blob/b1ec73be87889566e96b501171ff00004fee8c93/client/state/jetpack-connect/reducer/jetpack-connect-authorize.js#L76-L77). 
* Remove reducer and its corresponding unit test.
* Make note what action type name the reducer was listening to (`JETPACK_CONNECT_USER_ALREADY_CONNECTED`), locate corresponding action.
* Have notices middleware listen to action. If on mobile, dispatch a redirect action.

### Testing Instructions

See @sirreal's [comment](https://github.com/Automattic/wp-calypso/pull/24645#issuecomment-386254827) and the screenshots below. (You have to add `&calypso_env=development` to the JP Connect URL to test this branch.)

### References
(showing how the mobile redirect was implemented in `jetpack-connect/authorize`): 
https://github.com/Automattic/wp-calypso/blob/b1ec73be87889566e96b501171ff00004fee8c93/client/jetpack-connect/authorize.js#L376-L382

https://github.com/Automattic/wp-calypso/blob/b1ec73be87889566e96b501171ff00004fee8c93/client/jetpack-connect/authorize.js#L653-L654

### Screenshots

Before:

![image](https://user-images.githubusercontent.com/96308/39574767-facc6762-4ee0-11e8-8f98-4e01ee17499f.png)

After:

![image](https://user-images.githubusercontent.com/96308/39574658-acb373a4-4ee0-11e8-9ca2-2b157154a98c.png)
